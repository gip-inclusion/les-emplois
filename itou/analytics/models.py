import uuid

from django.db import models
from django.utils import timezone

from itou.common_apps.address.departments import DEPARTMENTS
from itou.users.enums import UserKind


class DatumCode(models.TextChoices):
    # Employee record - Base
    EMPLOYEE_RECORD_COUNT = "ER-001", "FS totales"
    EMPLOYEE_RECORD_DELETED = "ER-002", "FS (probablement) supprimées"
    # Employee record - Lifecycle
    EMPLOYEE_RECORD_PROCESSED_AT_FIRST_EXCHANGE = "ER-101", "FS intégrées (0000) au premier retour"
    EMPLOYEE_RECORD_WITH_ERROR_AT_FIRST_EXCHANGE = "ER-102", "FS avec une erreur au premier retour"
    EMPLOYEE_RECORD_WITH_ERROR_3436_AT_FIRST_EXCHANGE = "ER-102-3436", "FS avec une erreur 3436 au premier retour"
    EMPLOYEE_RECORD_WITH_AT_LEAST_ONE_ERROR = "ER-103", "FS ayant eu au moins un retour en erreur"
    # Approval - Base
    APPROVAL_COUNT = "AP-001", "PASS IAE total"
    APPROVAL_CANCELLED = "AP-002", "PASS IAE annulés"
    # Approval - PE notification cycle
    APPROVAL_PE_NOTIFY_SUCCESS = "AP-101", "PASS IAE synchronisés avec succès avec pole emploi"
    APPROVAL_PE_NOTIFY_PENDING = "AP-102", "PASS IAE en attente de synchronisation avec pole emploi"
    APPROVAL_PE_NOTIFY_ERROR = "AP-103", "PASS IAE en erreur de synchronisation avec pole emploi"
    APPROVAL_PE_NOTIFY_READY = "AP-104", "PASS IAE prêts à être synchronisés avec pole emploi"
    # Users
    USER_COUNT = "US-001", "Nombre d'utilisateurs"
    USER_JOB_SEEKER_COUNT = "US-011", "Nombre de demandeurs d'emploi"
    USER_PRESCRIBER_COUNT = "US-012", "Nombre de prescripteurs"
    USER_EMPLOYER_COUNT = "US-013", "Nombre d'employeurs"
    USER_LABOR_INSPECTOR_COUNT = "US-014", "Nombre d'inspecteurs du travail"
    USER_ITOU_STAFF_COUNT = "US-015", "Nombre d'administrateurs"
    # API usage
    API_TOTAL_CALLS = "API-001", "API : total d'appels reçus"
    API_TOTAL_UV = "API-002", "API : total de visiteurs uniques"
    API_TOTAL_CALLS_CANDIDATS = "API-003", "API candidats : total d'appels reçus"
    API_TOTAL_UV_CANDIDATS = "API-004", "API candidats : total de visiteurs uniques"
    API_TOTAL_CALLS_GEIQ = "API-005", "API GEIQ : total d'appels reçus"
    API_TOTAL_UV_GEIQ = "API-006", "API GEIQ : total de visiteurs uniques"
    API_TOTAL_CALLS_ER = "API-007", "API FS : total d'appels reçus"
    API_TOTAL_UV_ER = "API-008", "API FS : total de visiteurs uniques"
    API_TOTAL_CALLS_MARCHE = "API-009", "API Le marché : total d'appels reçus"
    API_TOTAL_CALLS_SIAES = "API-010", "API siaes : total d'appels reçus"
    API_TOTAL_UV_SIAES = "API-011", "API siaes : total de visiteurs uniques (par adresse IP)"
    API_TOTAL_CALLS_STRUCTURES = "API-012", "API structures : total d'appels reçus"


class Datum(models.Model):
    """Store an aggregated `value` of the `code` data point for the specified `bucket`."""

    id = models.UUIDField(primary_key=True, default=uuid.uuid4)

    code = models.TextField(choices=DatumCode.choices)
    bucket = models.TextField()
    value = models.IntegerField()  # Integer offers the best balance between range, storage size, and performance

    measured_at = models.DateTimeField(default=timezone.now)  # Not using auto_now_add=True to allow overrides

    class Meta:
        verbose_name_plural = "data"
        unique_together = ["code", "bucket"]
        indexes = [models.Index(fields=["measured_at", "code"])]


class StatsDashboardVisit(models.Model):
    dashboard_id = models.IntegerField(verbose_name="ID tableau de bord Metabase")
    dashboard_name = models.TextField(verbose_name="nom de la vue du tableau de bord")
    department = models.CharField(verbose_name="département", choices=DEPARTMENTS.items(), max_length=3, null=True)
    region = models.TextField(verbose_name="région", null=True)
    current_company_id = models.IntegerField(verbose_name="ID entreprise courante", null=True)
    current_prescriber_organization_id = models.IntegerField(
        verbose_name="ID organisation prescriptrice courante", null=True
    )
    current_institution_id = models.IntegerField(verbose_name="ID institution courante", null=True)
    user_kind = models.TextField(verbose_name="type d'utilisateur", choices=UserKind.choices)
    user_id = models.IntegerField(verbose_name="ID utilisateur")

    measured_at = models.DateTimeField(default=timezone.now)  # Not using auto_now_add=True to allow overrides

    class Meta:
        verbose_name_plural = "visite de tableau de bord"
