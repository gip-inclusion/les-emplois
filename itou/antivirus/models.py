from django.db import models
from django.utils.text import capfirst

from itou.files.models import File


class Scan(models.Model):
    file = models.OneToOneField(File, on_delete=models.CASCADE)
    clamav_signature = models.TextField()
    clamav_completed_at = models.DateTimeField(null=True, verbose_name="analyse ClamAV le")
    clamav_infected = models.BooleanField(null=True, verbose_name="fichier infect√© selon ClamAV")
    comment = models.TextField(blank=True, verbose_name="commentaire")

    class Meta:
        verbose_name = "analyse antivirus"
        verbose_name_plural = "analyses antivirus"

    def __str__(self):
        text = f"{capfirst(self._meta.verbose_name)} {self.file_id}"
        if self.clamav_infected:
            text = f"[VIRUS] {text}"
        return text
