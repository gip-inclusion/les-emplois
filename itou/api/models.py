import secrets

from django.db import models
from django.utils import timezone

from itou.companies.models import Company


def _generate_key():
    return secrets.token_urlsafe()


class CompanyToken(models.Model):
    key = models.CharField(default=_generate_key, unique=True, editable=False)
    label = models.CharField(verbose_name="m√©mo permettant d'identifier l'usage du jeton", max_length=60, unique=True)
    created_at = models.DateTimeField(default=timezone.now)
    companies = models.ManyToManyField(Company, related_name="api_tokens")

    class Meta:
        verbose_name = "jeton d'API entreprise"
        verbose_name_plural = "jetons d'API entreprise"
