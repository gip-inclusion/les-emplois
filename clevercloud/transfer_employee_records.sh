#!/bin/bash -l

set -u

if [[ "$CRON_ENABLED" != "1" ]]; then
    echo "Crons not enabled."
    exit 0
fi

if [[ "$EMPLOYEE_RECORD_CRON_ENABLED" != "1" ]]; then
    exit 0
fi

# About clever cloud cronjobs:
# https://www.clever-cloud.com/doc/administrate/cron/#deduplicating-crons
if [[ "$INSTANCE_NUMBER" != "0" ]]; then
    echo "Instance number is ${INSTANCE_NUMBER}. Stop here."
    exit 0
fi

cd "$APP_HOME" || exit
django-admin transfer_employee_records --wet-run "$@"
django-admin transfer_employee_records_updates --wet-run "$@"
