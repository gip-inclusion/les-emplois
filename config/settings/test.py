import os


ITOU_ENVIRONMENT = "DEV"
os.environ["ITOU_ENVIRONMENT"] = ITOU_ENVIRONMENT

from .base import *  # noqa: E402,F403


SECRET_KEY = "foobar"
ALLOWED_HOSTS = []

# We *want* to do the same `collectstatic` on the CI than on PROD to catch errors early,
# but we don't want to do it when running the test suite locally for performance reasons.
if not os.getenv("CI", False):
    # `ManifestStaticFilesStorage` (used in base settings) requires `collectstatic` to be run.
    STORAGES["staticfiles"]["BACKEND"] = "django.contrib.staticfiles.storage.StaticFilesStorage"  # noqa: F405

ASP_ITOU_PREFIX = "XXXXX"  # same as in our fixtures
ITOU_PROTOCOL = "http"
ITOU_FQDN = "localhost:8000"

DATABASES["default"]["HOST"] = os.getenv("PGHOST", "127.0.0.1")  # noqa: F405
DATABASES["default"]["PORT"] = os.getenv("PGPORT", "5432")  # noqa: F405
DATABASES["default"]["NAME"] = os.getenv("PGDATABASE", "itou")  # noqa: F405
DATABASES["default"]["USER"] = os.getenv("PGUSER", "postgres")  # noqa: F405
DATABASES["default"]["PASSWORD"] = os.getenv("PGPASSWORD", "password")  # noqa: F405

MAILJET_API_KEY_PRINCIPAL = "API_MAILJET_KEY_PRINCIPAL"
MAILJET_SECRET_KEY_PRINCIPAL = "API_MAILJET_SECRET_PRINCIPAL"

AWS_S3_ENDPOINT_URL = f"http://{os.getenv('CELLAR_ADDON_HOST', 'localhost:9000')}/"
AWS_S3_ACCESS_KEY_ID = "minioadmin"
AWS_S3_SECRET_ACCESS_KEY = "minioadmin"
AWS_STORAGE_BUCKET_NAME = "tests"
