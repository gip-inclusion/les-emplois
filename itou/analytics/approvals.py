from django.db.models import Count, Q

from itou.approvals import models as approvals_models
from itou.utils.apis import enums as api_enums

from . import models


def collect_approvals_counts(before):
    counts = approvals_models.Approval.objects.filter(created_at__lt=before).aggregate(
        total=Count("pk"),
        pe_notify_success=Count("pk", filter=Q(pe_notification_status=api_enums.PEApiNotificationStatus.SUCCESS)),
        pe_notify_pending=Count(
            "pk",
            filter=Q(
                pe_notification_status__in=(
                    api_enums.PEApiNotificationStatus.PENDING,
                    api_enums.PEApiNotificationStatus.SHOULD_RETRY,
                )
            ),
        ),
        pe_notify_error=Count("pk", filter=Q(pe_notification_status=api_enums.PEApiNotificationStatus.ERROR)),
    )
    return {
        models.DatumCode.APPROVAL_COUNT: counts["total"],
        models.DatumCode.APPROVAL_PE_NOTIFY_SUCCESS: counts["pe_notify_success"],
        models.DatumCode.APPROVAL_PE_NOTIFY_PENDING: counts["pe_notify_pending"],
        models.DatumCode.APPROVAL_PE_NOTIFY_ERROR: counts["pe_notify_error"],
    }


def collect_cancelled_approvals(before):
    cancelled_count = approvals_models.CancelledApproval.objects.filter(created_at__lt=before).count()
    return {models.DatumCode.APPROVAL_CANCELLED: cancelled_count}


def collect_analytics_data(before):
    return {
        **collect_approvals_counts(before),
        **collect_cancelled_approvals(before),
    }
