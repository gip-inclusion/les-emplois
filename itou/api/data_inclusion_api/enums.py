import enum


class StructureTypeStr(str, enum.Enum):
    ORGA = "orga"
    SIAE = "siae"
