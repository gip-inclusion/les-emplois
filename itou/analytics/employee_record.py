from django.db.models import Count, Max, Min

from itou.analytics import models
from itou.employee_record import models as employee_record_models


def collect_employee_records_count(before):
    return {
        models.DatumCode.EMPLOYEE_RECORD_COUNT: employee_record_models.EmployeeRecord.objects.filter(
            created_at__lt=before
        ).count(),
    }


def collect_probably_deleted_employee_records(before):
    data = employee_record_models.EmployeeRecord.objects.filter(created_at__lt=before).aggregate(
        first_pk=Min("pk"),
        last_pk=Max("pk"),
        count_pk=Count("pk"),
    )

    value = ((data["last_pk"] - data["first_pk"] + 1) - data["count_pk"]) if None not in data.values() else 0
    return {models.DatumCode.EMPLOYEE_RECORD_DELETED: value}


def collect_employee_records_processing_code_of_first_exchange(before):
    data = (
        employee_record_models.EmployeeRecord.objects.filter(created_at__lt=before)
        .values("asp_processing_code")
        .annotate(
            count=Count("asp_processing_code"),
        )
    )
    count_by_processing_code = {item["asp_processing_code"]: item["count"] for item in data}

    return {
        models.DatumCode.EMPLOYEE_RECORD_PROCESSED_AT_FIRST_EXCHANGE: count_by_processing_code.get(
            employee_record_models.EmployeeRecord.ASP_PROCESSING_SUCCESS_CODE, 0
        ),
        models.DatumCode.EMPLOYEE_RECORD_WITH_ERROR_AT_FIRST_EXCHANGE: sum(
            count
            for code, count in count_by_processing_code.items()
            if code != employee_record_models.EmployeeRecord.ASP_PROCESSING_SUCCESS_CODE
        ),
        models.DatumCode.EMPLOYEE_RECORD_WITH_ERROR_3436_AT_FIRST_EXCHANGE: count_by_processing_code.get(
            employee_record_models.EmployeeRecord.ASP_DUPLICATE_ERROR_CODE, 0
        ),
    }


def collect_employee_record_with_at_least_one_error(before):
    employee_records_with_at_least_one_error = employee_record_models.EmployeeRecordQuerySet.union(
        employee_record_models.EmployeeRecord.objects.filter(created_at__lt=before)
        .exclude(
            asp_processing_code=employee_record_models.EmployeeRecord.ASP_PROCESSING_SUCCESS_CODE,
        )
        .values("pk"),
        employee_record_models.EmployeeRecordUpdateNotification.objects.filter(created_at__lt=before)
        .exclude(
            asp_processing_code=employee_record_models.EmployeeRecord.ASP_PROCESSING_SUCCESS_CODE,
        )
        .values("employee_record"),
    )

    value = (
        employee_record_models.EmployeeRecord.objects.filter(pk__in=employee_records_with_at_least_one_error)
        .distinct()
        .count()
    )
    return {models.DatumCode.EMPLOYEE_RECORD_WITH_AT_LEAST_ONE_ERROR: value}


def collect_analytics_data(before):
    return {
        **collect_employee_records_count(before),
        **collect_probably_deleted_employee_records(before),
        **collect_employee_records_processing_code_of_first_exchange(before),
        **collect_employee_record_with_at_least_one_error(before),
    }
