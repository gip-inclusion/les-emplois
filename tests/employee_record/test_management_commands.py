from unittest import mock

import freezegun
import pytest
from django.test.utils import override_settings

from itou.employee_record.enums import Status
from itou.employee_record.mocks.transfer_employee_records import (
    SFTPAllDupsConnectionMock,
    SFTPBadConnectionMock,
    SFTPConnectionMock,
    SFTPEvilConnectionMock,
    SFTPGoodConnectionMock,
)
from itou.employee_record.models import EmployeeRecord
from itou.utils.mocks.address_format import mock_get_geocoding_data
from tests.approvals.factories import ProlongationFactory, SuspensionFactory
from tests.employee_record.common import ManagementCommandTestCase
from tests.job_applications.factories import JobApplicationWithCompleteJobSeekerProfileFactory


# There is no need to create 700 employee records for a single batch
# so this class var is changed to 1 for tests, otherwise download operation is not triggered.
@mock.patch("itou.employee_record.models.EmployeeRecordBatch.MAX_EMPLOYEE_RECORDS", new=1)
@override_settings(ASP_FS_SFTP_HOST="foobar.com", ASP_FS_SFTP_USER="django_tests")
class EmployeeRecordManagementCommandTest(ManagementCommandTestCase):
    """
    Employee record management command, testing:
    - mocked sftp connection
    - basic upload / download modes
    - ...
    """

    MANAGEMENT_COMMAND_NAME = "transfer_employee_records"

    @classmethod
    @mock.patch(
        "itou.common_apps.address.format.get_geocoding_data",
        side_effect=mock_get_geocoding_data,
    )
    def setUpTestData(cls, _mock):
        cls.job_application = JobApplicationWithCompleteJobSeekerProfileFactory()
        cls.employee_record = EmployeeRecord.from_job_application(cls.job_application)
        cls.employee_record.update_as_ready()

    @mock.patch("pysftp.Connection", SFTPConnectionMock)
    def test_smoke_download(self):
        out, _ = self.call_command(download=True)
        assert out == self.snapshot

    @mock.patch("pysftp.Connection", SFTPConnectionMock)
    def test_smoke_upload(self):
        with freezegun.freeze_time("2021-09-27"):
            out, _ = self.call_command(upload=True)
        assert out == self.snapshot

    @mock.patch("pysftp.Connection", SFTPConnectionMock)
    def test_smoke_download_and_upload(self):
        with freezegun.freeze_time("2021-09-27"):
            out, _ = self.call_command(upload=True, download=True)
        assert out == self.snapshot

    @mock.patch("pysftp.Connection", SFTPGoodConnectionMock)
    @mock.patch(
        "itou.common_apps.address.format.get_geocoding_data",
        side_effect=mock_get_geocoding_data,
    )
    def test_dry_run_upload_and_download(self, _mock):
        self.call_command(upload=True, download=True, dry_run=True)

        self.employee_record.refresh_from_db()
        assert self.employee_record.status == Status.READY

    @mock.patch("pysftp.Connection", SFTPBadConnectionMock)
    def test_upload_failure(self):
        with pytest.raises(Exception):
            self.call_command(upload=True)

        self.employee_record.refresh_from_db()
        assert self.employee_record.status == Status.READY

    @mock.patch("pysftp.Connection", SFTPBadConnectionMock)
    def test_download_failure(self):
        with pytest.raises(Exception):
            self.call_command(download=True)

        self.employee_record.refresh_from_db()
        assert self.employee_record.status == Status.READY

    @mock.patch("pysftp.Connection", SFTPEvilConnectionMock)
    @mock.patch(
        "itou.common_apps.address.format.get_geocoding_data",
        side_effect=mock_get_geocoding_data,
    )
    def test_random_connection_failure(self, _mock):
        employee_record = self.employee_record

        # Random upload failure
        with pytest.raises(Exception):
            self.call_command(upload=True)
        # Employee record must be in the same status
        employee_record.refresh_from_db()
        assert employee_record.status == Status.READY

        with pytest.raises(Exception):
            self.call_command(download=True)
        # Employee record must be in the same status
        employee_record.refresh_from_db()
        assert employee_record.status == Status.READY

    @mock.patch("pysftp.Connection", SFTPGoodConnectionMock)
    @mock.patch(
        "itou.common_apps.address.format.get_geocoding_data",
        side_effect=mock_get_geocoding_data,
    )
    def test_upload_and_download_success(self, _mock):
        """
        - Create an employee record
        - Send it to ASP
        - Get feedback
        - Update employee record
        """
        employee_record = self.employee_record

        with freezegun.freeze_time("2021-09-27"):
            out, _ = self.call_command(upload=True)
        assert out == self.snapshot(name="upload")

        employee_record.refresh_from_db()
        assert employee_record.status == Status.SENT
        assert employee_record.asp_batch_line_number == 1
        assert employee_record.asp_batch_file is not None

        out, _ = self.call_command(download=True)
        assert out == self.snapshot(name="download")

        employee_record.refresh_from_db()
        assert employee_record.status == Status.PROCESSED
        assert employee_record.asp_processing_code == EmployeeRecord.ASP_PROCESSING_SUCCESS_CODE
        assert (
            employee_record.archived_json.get("libelleTraitement")
            == "La ligne de la fiche salarié a été enregistrée avec succès."
        )

    @mock.patch("pysftp.Connection", SFTPGoodConnectionMock)
    def test_asp_test(self):
        out, _ = self.call_command(test=True, download=True)
        assert out == self.snapshot

    @override_settings(ASP_FS_SFTP_HOST="")
    def test_wrong_environment(self):
        out, _ = self.call_command(download=True)
        assert out == self.snapshot

    @mock.patch("pysftp.Connection", SFTPAllDupsConnectionMock)
    @mock.patch(
        "itou.common_apps.address.format.get_geocoding_data",
        side_effect=mock_get_geocoding_data,
    )
    def test_automatic_duplicates_processing(self, _):
        # Check that from now on employee records with a 3436 processing code
        # are auto-magically converted as PROCESSED employee records

        # Don't forget to make a complete upload / download cycle
        self.call_command(upload=True, download=True)

        self.employee_record.refresh_from_db()

        assert 0 == EmployeeRecord.objects.asp_duplicates().count()
        assert Status.PROCESSED == self.employee_record.status
        assert self.employee_record.processed_as_duplicate

    @mock.patch("pysftp.Connection", SFTPAllDupsConnectionMock)
    @mock.patch(
        "itou.common_apps.address.format.get_geocoding_data",
        side_effect=mock_get_geocoding_data,
    )
    def test_duplicates_with_a_suspension_generate_an_update_notification(self, _):
        SuspensionFactory(approval=self.employee_record.job_application.approval)

        # Don't forget to make a complete upload / download cycle
        self.call_command(upload=True, download=True)

        self.employee_record.refresh_from_db()

        assert self.employee_record.update_notifications.count() == 1

    @mock.patch("pysftp.Connection", SFTPAllDupsConnectionMock)
    @mock.patch(
        "itou.common_apps.address.format.get_geocoding_data",
        side_effect=mock_get_geocoding_data,
    )
    def test_duplicates_with_a_prolongation_generate_an_update_notification(self, _):
        ProlongationFactory(approval=self.employee_record.job_application.approval)

        # Don't forget to make a complete upload / download cycle
        self.call_command(upload=True, download=True)

        self.employee_record.refresh_from_db()

        assert self.employee_record.update_notifications.count() == 1
