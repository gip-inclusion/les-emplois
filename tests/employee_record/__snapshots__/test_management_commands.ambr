# serializer version: 1
# name: EmployeeRecordManagementCommandTest.test_asp_test
  '''
  Using *TEST* JSON serializers (SIRET number mapping)
  Connected to "foobar.com" as "django_tests"
  Current remote dir is "PWD"
  Starting DOWNLOAD of feedback files
  Successfully parsed 0/0 files
  Employee records processing done!
  
  '''
# ---
# name: EmployeeRecordManagementCommandTest.test_smoke_download
  '''
  Connected to "foobar.com" as "django_tests"
  Current remote dir is "PWD"
  Starting DOWNLOAD of feedback files
  Successfully parsed 0/0 files
  Employee records processing done!
  
  '''
# ---
# name: EmployeeRecordManagementCommandTest.test_smoke_download_and_upload
  '''
  Connected to "foobar.com" as "django_tests"
  Current remote dir is "PWD"
  Starting UPLOAD of employee records
  Successfully uploaded: RIAE_FS_20210927000000.json
  Starting DOWNLOAD of feedback files
  Fetching file: RIAE_FS_20210927000000_FichierRetour.json
  Record: line_number=1, processing_code=None, processing_label=None
  Successfully processed 'RIAE_FS_20210927000000_FichierRetour.json', it can be deleted.
  Deleting 'RIAE_FS_20210927000000_FichierRetour.json' from SFTP server
  Successfully parsed 1/1 files
  Employee records processing done!
  
  '''
# ---
# name: EmployeeRecordManagementCommandTest.test_smoke_upload
  '''
  Connected to "foobar.com" as "django_tests"
  Current remote dir is "PWD"
  Starting UPLOAD of employee records
  Successfully uploaded: RIAE_FS_20210927000000.json
  Employee records processing done!
  
  '''
# ---
# name: EmployeeRecordManagementCommandTest.test_upload_and_download_success[download]
  '''
  Connected to "foobar.com" as "django_tests"
  Current remote dir is "PWD"
  Starting DOWNLOAD of feedback files
  Fetching file: RIAE_FS_20210927000000_FichierRetour.json
  Record: line_number=1, processing_code='0000', processing_label='La ligne de la fiche salarié a été enregistrée avec succès.'
  Successfully processed 'RIAE_FS_20210927000000_FichierRetour.json', it can be deleted.
  Deleting 'RIAE_FS_20210927000000_FichierRetour.json' from SFTP server
  Successfully parsed 1/1 files
  Employee records processing done!
  
  '''
# ---
# name: EmployeeRecordManagementCommandTest.test_upload_and_download_success[upload]
  '''
  Connected to "foobar.com" as "django_tests"
  Current remote dir is "PWD"
  Starting UPLOAD of employee records
  Successfully uploaded: RIAE_FS_20210927000000.json
  Employee records processing done!
  
  '''
# ---
# name: EmployeeRecordManagementCommandTest.test_wrong_environment
  '''
  Your environment is missing ASP_FS_SFTP_HOST to run this command.
  
  '''
# ---
